Best practices/React/Variable naming.md

```diff
@@ -0,0 +1,121 @@
+# Variable naming
+
+### Abbreviations
+
+In order to be consistent with GraphQL arguments looking for ids, never capitalize the `D` in `id` when the referring to a GraphQL generated id. Use either `<noun>Id` in camel case or simply `id`.
+
+```javascript
+// bad
+const productID = product.id;
+
+// good
+const productId = product.id;
+```
+
+All other abbreviations should be named with all uppercase letters unless the variable starts with the abbreviation, in which case it should be all lowercase.
+
+```javascript
+// bad
+const SKU = product.sku;
+const SKULabel = product.sku;
+const resourceSku = resource.sku;
+
+// good
+const sku = product.sku;
+const skuLabel = 'SKU';
+const resourceSKU = resource.sku;
```

Comment:
I think this is probably wrong given how our pascal case requiring rules will make this `ResourceSku` instead